### PR TITLE
fix(v2): retry compaction polling requests on Raft leader errors

### PR DESCRIPTION
**Pull request by aleks-p**
Addresses an issue where compaction workers continue making requests to the wrong Raft node following a leadership transfer. We could alternatively move this to `applyCommand` but I will leave that for a future iteration.

### CODE REVIEW

**Review by korniltsev — APPROVED** [review]
lgtm.
While the fix is valid, I would expect the client to still be able to work without raft details - by pure random node selection. I will need to look into why it did not work.

File: pkg/experiment/metastore/metastore_state_poll_compaction_jobs.go [hunk 1]
```diff
@@ -23,6 +23,10 @@ func (m *Metastore) PollCompactionJobs(_ context.Context, req *compactorv1.PollC
 		"raft_last_index", m.raft.LastIndex(),
 		"raft_applied_index", m.raft.AppliedIndex())
 	_, resp, err := applyCommand[*compactorv1.PollCompactionJobsRequest, *compactorv1.PollCompactionJobsResponse](m.raft, req, m.config.Raft.ApplyTimeout)
+	if err != nil {
+		_ = level.Error(m.logger).Log("msg", "failed to apply poll compaction jobs", "raft_commit_index", m.raft.CommitIndex(), "err", err)
+		return nil, wrapRetryableErrorWithRaftDetails(err, m.raft)
+	}
```
Comment (kolesnikovae) [inline pkg/experiment/metastore/metastore_state_poll_compaction_jobs.go:29]: I guess we'll be facing this issue constantly – should we move this wrapper into the `applyCommand` handler?

**Review by kolesnikovae — APPROVED** [review]
LGTM!

File: pkg/experiment/metastore/metastore_errors.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+func wrapRetryableErrorWithRaftDetails(err error, raft *raft.Raft) error {
+	if err != nil && shouldRetryCommand(err) {
+		_, serverID := raft.LeaderWithID()
+		s := status.New(codes.Unavailable, err.Error())
+		if serverID != "" {
+			s, _ = s.WithDetails(&typesv1.RaftDetails{Leader: string(serverID)})
+		}
+		return s.Err()
+	}
+	return err
```
[the remark this answers is not in the document]
Comment (aleks-p) [reply pkg/experiment/metastore/metastore_errors.go:22]: I agree in general. The downside is that the negation in `!shouldRetryCommand()` is usually harder to read (easier to miss), but I could try to rename the method instead.

**Pull request merged**